src/web/components/Core/PagingTool.tsx

```diff
@@ -0,0 +1,160 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useEffect, useState } from 'react';
+
+import { SelectDropdown, SelectOption } from './SelectDropdown';
+
+import './PagingTool.scss';
+
+const rowsPerPageValues = [10, 25, 50, 100, 250] as const;
+
+export type RowsPerPageValues = (typeof rowsPerPageValues)[number];
+
+type PagingProps = Readonly<{
+  numberTotalRows: number;
+  onChangeRows: (currentPageNumber: number, currentRowsPerPage: RowsPerPageValues) => void;
+  rowsPerPageTitle?: string;
+  initialRowsPerPage?: RowsPerPageValues;
+  initialPageNumber?: number;
+}>;
+
+export function PagingTool({
+  numberTotalRows,
+  onChangeRows,
+  rowsPerPageTitle = 'Rows Per Page',
+  initialRowsPerPage = 10,
+  initialPageNumber = 1,
+}: PagingProps) {
+  const initialPageNumberOptions = Array.from(
+    { length: Math.ceil(numberTotalRows / initialRowsPerPage) },
+    (_, index) => index + 1
+  ).map((number) => ({
+    name: number.toString(),
+    id: number,
+  }));
+
+  const [rowsPerPage, setRowsPerPage] = useState<RowsPerPageValues>(initialRowsPerPage);
+  const [pageNumber, setPageNumber] = useState<number>(initialPageNumber);
+  const [pageNumberOptions, setPageNumberOptions] =
+    useState<SelectOption<number>[]>(initialPageNumberOptions);
+
+  useEffect(() => {
+    setPageNumberOptions(
+      Array.from(
+        { length: Math.ceil(numberTotalRows / initialRowsPerPage) },
+        (_, index) => index + 1
+      ).map((number) => ({
+        name: number.toString(),
+        id: number,
+      }))
+    );
+    setPageNumber(initialPageNumber);
+    setRowsPerPage(initialRowsPerPage);
+  }, [numberTotalRows, initialRowsPerPage, initialPageNumber]);
+
+  const rowsPerPageOptions =
+    numberTotalRows > 10
+      ? rowsPerPageValues
+          .filter((value) => value <= numberTotalRows)
+          .map((value) => ({
+            name: value.toString(),
+            id: value,
+          }))
+      : [{ name: rowsPerPageValues[0].toString(), id: rowsPerPageValues[0] }];
```

Comment:
Making sure that there is at least one option for rows per page so it's not a blank dropdown even if there are only say 5 rows